jsonschema/validate.go

```diff
@@ -746,12 +746,23 @@ func structPropertiesOf(t reflect.Type) propertyMap {
 		return props.(propertyMap)
 	}
 	props := map[string]reflect.StructField{}
-	for _, sf := range reflect.VisibleFields(t) {
+	visit := func(sf reflect.StructField) error {
```

Comment:
Hmm, it looks like reflect.VisibleFields does exactly what you want, and there's no need for the new traversal.
https://pkg.go.dev/reflect#VisibleFields

So I think the only change necessary is to use VisibleFields in inference?

Reply:
Yup you're right! Updated!